Ginger/GingerCore/ALM/QCRestAPICore.cs

```diff
@@ -0,0 +1,115 @@
+        {
+            return ImportFromQCRest.ImportTestSetData(testSet);
+        }
+
+        public BusinessFlow ConvertQCTestSetToBF(QC.QCTestSet testSet)
+        {
+            return ImportFromQCRest.ConvertQCTestSetToBF(testSet);
+        }
+
+        public override Dictionary<Guid, string> CreateNewALMDefects(Dictionary<Guid, Dictionary<string, string>> defectsForOpening, bool useREST = false)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool DisconnectALMProjectStayLoggedIn()
+        {
+            return QCRestAPIConnect.DisconnectQCProjectStayLoggedIn();
+        }
+
+        public bool ExportActivitiesGroupToALM(ActivitiesGroup activtiesGroup, QCTestCase matchingTC, string uploadPath, ObservableList<ExternalItemFieldBase> testCaseFields, ObservableList<ExternalItemFieldBase> designStepsFields, ObservableList<ExternalItemFieldBase> designStepsParamsFields, ref string res)
+        {
+            return ExportToQCRestAPI.ExportActivitiesGroupToQC(activtiesGroup, matchingTC, uploadPath, testCaseFields, designStepsFields, designStepsParamsFields, ref res);
+        }
+
+        public bool ExportBusinessFlowToALM(BusinessFlow businessFlow, QCTestSet mappedTestSet, string uploadPath, ObservableList<ExternalItemFieldBase> testSetFields, ObservableList<ExternalItemFieldBase> testInstanceFields, ref string result)
+        {
+            return ExportToQCRestAPI.ExportBusinessFlowToQC(businessFlow, mappedTestSet, uploadPath, testSetFields, testInstanceFields, ref result);
+        }
+
+        public override bool ExportExecutionDetailsToALM(BusinessFlow bizFlow, ref string result, bool exectutedFromAutomateTab = false, PublishToALMConfig publishToALMConfig = null)
+        {
+            ObservableList<ExternalItemFieldBase> runFields = GetALMItemFields(null, true, ALM_Common.DataContracts.ResourceType.TEST_RUN);
+            return ExportToQCRestAPI.ExportExceutionDetailsToALM(bizFlow, ref result, runFields, exectutedFromAutomateTab, publishToALMConfig);
+        }
+
+        public override ObservableList<ActivitiesGroup> GingerActivitiesGroupsRepo
+        {
+            get { return ImportFromQCRest.GingerActivitiesGroupsRepo; }
+            set { ImportFromQCRest.GingerActivitiesGroupsRepo = value; }
+        }
+
+        public override ObservableList<Activity> GingerActivitiesRepo
+        {
+            get { return ImportFromQCRest.GingerActivitiesRepo; }
+            set { ImportFromQCRest.GingerActivitiesRepo = value; }
+        }
+
+        public override ObservableList<ExternalItemFieldBase> GetALMItemFields(BackgroundWorker bw, bool online, ALM_Common.DataContracts.ResourceType resourceType)
+        {
+            if (resourceType == ALM_Common.DataContracts.ResourceType.ALL)
+                return ImportFromQCRest.GetALMItemFields();
+            return ImportFromQCRest.GetALMItemFields(resourceType);
```

Comment:
Why not to always pass "resourceType" and let ImportFromQCRest.GetALMItemFields() to handle also the case of ResourceType.ALL instead of this check and 2 diffrent calls